desmod/component.py

```diff
@@ -275,6 +275,23 @@ def elab_hook(self):
         """
         pass
 
+    def post_fail(self):
+        """Recursively run post-fail hooks."""
+        for child in self._children:
+            child.post_fail()
+        self.post_sim_fail_hook()
+
+    def post_sim_fail_hook(self):
```

Comment:
Do you mean, replace `post_sim_fail_hook()` with `on_exception()` and `post_fail()` with `on_exception()`?